pkg/operator/encryption/state/helpers.go

```diff
@@ -34,11 +34,15 @@ func MigratedFor(grs []schema.GroupResource, km KeyState) (ok bool, missing []sc
 	return true, nil, ""
 }
 
-// KeysWithPotentiallyPersistedData returns the minimal, recent secrets which have migrated all given GRs.
-func KeysWithPotentiallyPersistedData(grs []schema.GroupResource, recentFirstSortedKeys []KeyState) []KeyState {
+// KeysWithPotentiallyPersistedDataAndNextReadKey returns the minimal, recent secrets which have migrated all given GRs.
+func KeysWithPotentiallyPersistedDataAndNextReadKey(grs []schema.GroupResource, recentFirstSortedKeys []KeyState) []KeyState {
 	for i, k := range recentFirstSortedKeys {
 		if allMigrated, missing, _ := MigratedFor(grs, k); allMigrated {
-			return recentFirstSortedKeys[:i+1]
+			if i+1 < len(recentFirstSortedKeys) {
```

Comment:
this looks to be safe, we could have a unit test for it.

Reply:
what do you want to test?

Reply:
we have tests that "one more read-key" is fine. So this is covered. The rest here is just Golang standard behaviour.

Reply:
sgtm.